### DISCUSSION TITLE
router.replace("/path") does not set cookie whereas window.location.replace does

### DISCUSSION THREAD

**Discussion by baoskee** [vercel/next.js#29247] [Help]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

15.13.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next dev

### Describe the Bug

router.replace("/path") does not set cookie whereas window.location.replace does. 

### Expected Behavior

router from useRouter() should set all cookies for the GET request on the cookie's path when callling `replace("/path")` or `push("/path")`

### To Reproduce

# In the body of a React component
```
    const router = useRouter()
    const [loaded, setLoaded] = useState(false)
    
    useEffect(() => {
        (async () => {
            if (!router.isReady)
                return 
            
            const res = await post<ApiRequestBody, ApiResponseBody>('http://localhost:3000/api/auth', {
                magic_token: router.query.token as string
            })

            // Redirect
            if (res.status != 200) 
                return setLoaded(true)

            /// WARNING: router.replace(:) does not send cookies! 
            return window.location.replace('/app')
        })()

    }, [router.isReady])
```

**Comment by ijjk** [comment]
Hi, this appears to be working correctly in `v11.1.2` using `res.setHeader('set-cookie', 'hello=world; Path=/')` in an API route. Are you seeing the cookie header in the fetch request done during `router.replace`?
